src/Razor/src/Microsoft.NET.Sdk.Razor/DiscoverMvcApplicationParts.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class DiscoverMvcApplicationParts : Task
+    {
+        [Required]
+        public ITaskItem[] MvcAssemblyNames { get; set; }
+
+        [Required]
+        public ITaskItem[] ResolvedReferences { get; set; }
+
+        [Output]
+        public string[] ApplicationPartAssemblyNames { get; set; }
+
+        public override bool Execute()
+        {
+            var referenceItems = new List<ResolveReferenceItem>();
+            foreach (var item in ResolvedReferences)
+            {
+                string assemblyName = null;
+                var fusionName = item.GetMetadata("FusionName");
```

Comment:
I haven't seen a scenario where `FusionName` is null, but based on this comment - https://github.com/microsoft/msbuild/blob/1e574340ca00a71678d7eb67f3f3e68d981fd994/src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs#L157-L159, it looks it's optional. We could get rid of the branching and make it required if it turns out `ReferencePath` always includes this metadata.

Reply:
That comment is for the input to RAR and you are looking at the output. I don't know about a situation where RAR would not have set it and the code you're calling (taken from msbuild) would find it. @rainersigwald again for the definitive answer.

Reply:
Great. I'm all for removing code 👍

Reply:
Agreed, looks to me like we should always set FusionName for RAR outputs. Delete the code 🎉